### DISCUSSION TITLE
the solution.py file in /trame-tutorial/04_application/ path can't read my .vtu file

### DISCUSSION THREAD

**Discussion by zhangfengeo** [Kitware/trame#218] [Q&A] [answered]
I want to import a vtu file to the browser,but when i execute the solution.py in the /trame-tutorial/04_application path, the CMD show errors,the errors show: "vtkDataArray.cxx:1062   ERR| vtkDoubleArray (00000121D6210AE0): The number of components do not match the number requested: 3 != 1".
I dont't konw where the "1" in the "3 != 1" come from, do I need to rewrite the solution.py file to resolve this problem yet?

**Comment by jourdain** [comment, marked as answer]
The example create a pipeline (contour) which expect a scalar (1) while your file seems to only have a vector (3).
You can try removing the VTK pipeline so no processing will happen on an invalid dataset.

**Reply by zhangfengeo** [reply]
Thank you! If I want to open a vtu file which contains vector(3) dataarrys, what  should i do to make the componets and magnitude shown in the layout?

**Reply by jourdain** [reply]
Is your question related to "how to add a drop down the the UI" or "how to control VTK to do it" or "both" or "else"?

- If it is the UI part you can follow [course](https://kitware.github.io/trame/docs/course_intro.html) or [the tutorial](https://kitware.github.io/trame/docs/tutorial.html) which will walk you through the incremental build of such UI and you should be able to reuse that logic elsewhere.
- If it is VTK knowledge, you can learn more with the [vtk-example website](https://kitware.github.io/vtk-examples/site/) and [books](https://vtk.org/documentation/).
- If it is both then you have the previous links to look at.
- Otherwise you will need to be more explicit

Finally if you are looking for help in building such application, you can reach out to [Kitware for support](https://www.kitware.com/contact/).

**Reply by zhangfengeo** [reply]
Thanks, i'm a novice on UI and VTK,think i need to  know more about the knowledge of vtk.
